core/store/migrations/0024_add_cron_spec_id.go

```diff
@@ -0,0 +1,44 @@
+package migrations
+
+import (
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+)
+
+const up24 = `
+		CREATE TABLE cron_specs (
+			id BIGSERIAL PRIMARY KEY,
+			cron_schedule text NOT NULL,
+			created_at timestamp with time zone NOT NULL,
+			updated_at timestamp with time zone NOT NULL
+		);
+
+		ALTER TABLE jobs ADD COLUMN cron_spec_id INT REFERENCES cron_specs(id),
+		DROP CONSTRAINT chk_only_one_spec,
+		ADD CONSTRAINT chk_only_one_spec CHECK (
+			num_nonnulls(offchainreporting_oracle_spec_id, direct_request_spec_id, flux_monitor_spec_id, keeper_spec_id, cron_spec_id) = 1
+		);
+	`
+
+const down24 = `
```

Comment:
probably want a simple up/down test in migrate_test.go to actually run this

Reply:
added test ~